File: Version 2.0/menstrual_app.py
import datetime
from menstrual_app_functions import get_date_of_current_menstrual_cycle_end_date, get_date_of_end_of_flow_day, get_start_date_of_ovulation, get_end_date_of_ovulation, get_safe_day_start_after_ovulation

print("WELCOME TO YRSD BOO TING")

yyyy = int(input("Enter the year(20XX): "))
mm = int(input("Enter the month your cycle started(Jan=>1): "))
dd = int(input("Enter the day your cycle started(1-31): "))
cycle_duration = int(input("Enter your average cycle length (21-35 days): "))
flow_duration = int(input("Enter your average flow length (2-10 days): "))

date_that_cycle_started = datetime.date(yyyy, mm, dd)

current_menstrual_end_date = get_date_of_current_menstrual_cycle_end_date(date_that_cycle_started, cycle_duration)
date_of_ovulation_end = get_end_date_of_ovulation(current_menstrual_end_date)

print("Your current cycle will end on the" , current_menstrual_end_date)
print("Your flow day is predicated to end on the" , get_date_of_end_of_flow_day(date_that_cycle_started, flow_duration))
print("Your ovulation start date is predicated to be", get_start_date_of_ovulation(current_menstrual_end_date),"to", get_end_date_of_ovulation(current_menstrual_end_date))
print("Safe Period before Ovulation:", date_that_cycle_started, "to", get_date_of_end_of_flow_day(date_that_cycle_started, flow_duration))
print("Safe Period after Ovulation:", get_safe_day_start_after_ovulation(date_of_ovulation_end), "to",current_menstrual_end_date)

File: daySeven/menstrualWahala/MenstrualApp.java
import java.util.Scanner;
import java.time.LocalDate;


public class MenstrualApp{

public static void main(String[] args){

Scanner input = new Scanner(System.in);
LocalDate date = LocalDate.now();
System.out.println("WELCOME TO YRSD BOO TING");


int year = date.getYear();
System.out.print("Enter the month your cycle started (Jan=>1): ");
int month = input.nextInt();
System.out.print("Enter the day your cycle started(1-31): ");
int day = input.nextInt();
System.out.print("What is your average cycle duration: ");
int cycleDurationLength = input.nextInt();
System.out.print("What is your average period days: ");
int flowDaysDuration = input.nextInt();



LocalDate dateOfStartOfMenstrualCycle = LocalDate.of(year, month, day);
LocalDate endOfCycle = MenstrualAppFunctions.getDateOfEndOfMenstrualCycle(dateOfStartOfMenstrualCycle, cycleDurationLength);

int loopCondition = 0;

while(loopCondition!=1){
String menuDisplay = """
What would you like to know?

1) End of your cycle
2) End of your flow
3) Ovulation Window
4) Safe Days
0) Exit

Enter your choice: """;

System.out.print(menuDisplay);

int navigator = input.nextInt();

switch(navigator){
	case 1:
		System.out.println("The Day your cycle  is predicated to end is : " + endOfCycle);
		break;
	case 2:
		System.out.println("The Day your period flow is predicated to end is : " + MenstrualAppFunctions.getEndOfFlowDays(dateOfStartOfMenstrualCycle, flowDaysDuration));
		break;
	case 3:
		try{
			System.out.println("Your ovulation window is predicated to be between: " + MenstrualAppFunctions.getFertileWindowStartDate(endOfCycle) + " to " + 	
			MenstrualAppFunctions.getFertileWindowEndDate(endOfCycle));
		}
		catch(NullPointerException npe){
			System.out.println("\nPlease seek Medical Assistance.\n");
		}
		break;
	case 4:
		try{
			System.out.println("Your Safe days before your Ovulation Window are between: " + dateOfStartOfMenstrualCycle + " to " + 	MenstrualAppFunctions.getEndOfFlowDays(dateOfStartOfMenstrualCycle, flowDaysDuration) +"\nYour Safe days after your Ovulation Window are between: "+ 	MenstrualAppFunctions.getStartOfSafeDaysAfterFertileWindow(endOfCycle) +" to "+ endOfCycle);
		}
		catch(NullPointerException npe){
			System.out.println("\nPlease seek Medical Assistance.\n");
		}

		break;
	case 0:
		System.out.print("Good bye, boo.");
		loopCondition=1;
		break;
	default:
		System.out.print("INVALID INPUT!");
}



}

}}